src/System.Net.Security/src/System/Net/CertificateValidationPal.Windows.cs

```diff
@@ -187,12 +186,12 @@ internal static X509Store EnsureStoreOpened(bool isMachineStore)
         {
             X509Store store = isMachineStore ? s_myMachineCertStoreEx : s_myCertStoreEx;
 
-            if (store == null)
+            if (Volatile.Read(ref store) == null)
             {
                 lock (s_syncObject)
                 {
                     store = isMachineStore ? s_myMachineCertStoreEx : s_myCertStoreEx;
-                    if (store == null)
+                    if (Volatile.Read(ref store) == null)
```

Comment:
Various articles mention that the variable that's being checked needs to be `volatile`. To quote exactly:
> Without any memory barriers, it's broken in the ECMA CLI specification too. It's possible that under the .NET 2.0 memory model (which is stronger than the ECMA spec) it's safe, but I'd rather not rely on those stronger semantics, especially if there's any doubt as to the safety. Making the instance variable volatile can make it work, as would explicit memory barrier calls, although in the latter case even experts can't agree exactly which barriers are required. I tend to try to avoid situations where experts don't agree what's right and what's wrong!
Taken from: http://csharpindepth.com/Articles/General/Singleton.aspx

Most likely all instances of the double-checked lock mechanisms used throughout our code could be changed to either the `static` or the `Lazy<T>` implementations. I'll leave a `TODO` here and create a task.